UrlWithInformation into UrlInfo and StoredUrl

Yardenb opened the pull request:
Separated UrlInformation from StoredUrl.
Added a type UrlWithInformation that contains both UrlInformation and StoredUrl.
Changed the usage in some places from UrlWithInformation to UrlInformation.
I'm not sure if the usage changes are correct and if I missed places to replace.

SnirShechter commented on src/services/storage/drivers/relational/index.ts:
```diff
@@ -65,7 +65,7 @@ export class RelationalStorage implements StorageDriver {
 		constructor(public storage: RelationalStorage) {}
 
-		public async get(id: string, options = { withInfo: false }): Promise<StoredUrl | UrlWithInformation> {
+		public async get(id: string, options = { withInfo: false }): Promise<StoredUrl | UrlInformation> {
```
That's actually a UrlWithInformation.

the variable named urlInfo is the incorrect name here (it should be storedUrlWithInfo).

The story here is that the get function can return only the url, or the url with the info (for admins)

SnirShechter commented on src/services/storage/drivers/relational/index.ts:
```diff
@@ -111,12 +111,12 @@ export class RelationalStorage implements StorageDriver {
 		public async save(urlBody: UrlRequestData): Promise<StoredUrl> {
 			const url = urlBody.url
-			const urlInfo: Partial<UrlWithInformation> = {
+			const urlInfo: Partial<UrlInformation> = {
```
Can you remove the Partial<> here? It should be just UrlInformation

SnirShechter commented on src/services/storage/types/url.ts:
```diff
@@ -26,10 +26,11 @@ export interface UrlRequestData {
 
-export interface UrlWithInformation extends StoredUrl {
+export interface UrlInformation {
 	urlId: string
```
You can remove the urlId here.

SnirShechter commented on src/services/storage/index.ts:
```diff
@@ -39,7 +39,7 @@ export class Storage implements StorageDriver {
 		}
 
-		public async get(id: string, options = { withInfo: false }): Promise<StoredUrl | UrlWithInformation> {
+		public async get(id: string, options = { withInfo: false }): Promise<StoredUrl | UrlInformation> {
```
That's actually a UrlWithInformation.

SnirShechter commented on src/services/storage/types/url.ts:
```diff
@@ -6,7 +6,7 @@ export interface StoredUrl {
 
 export interface UrlStorageDriver {
-	get(id: string, options: { withInfo: boolean }): Promise<StoredUrl | UrlWithInformation>
+	get(id: string, options: { withInfo: boolean }): Promise<StoredUrl | UrlInformation>
```
That's actually a UrlWithInformation

SnirShechter commented:
There are probably some more places where you need to change the typings. Can you recheck?

Yardenb commented:
@SnirShechter Yeah, I didn't want to touch anything that was near a table, but now I'm thinking that's exactly what I was supposed to do... Thanks for the comments, I'm on it

SnirShechter commented:
Hehe, yea. I think you got it.

No problem, take your time 🙂

Yardenb commented:
Disclaimer for the last commit: I only tested it with inMemory storage. When I tried to run the relational storage I got an error regarding one of the migrations files. I don't think I caused this error, because I get the exact error when I run the original project, but I just wanted to give you a heads up about it.

SnirShechter commented on src/services/storage/drivers/relational/index.ts:
```diff
@@ -111,32 +111,32 @@ export class RelationalStorage implements StorageDriver {
 				infoVisitCount: 0,
 				lastUsed: new Date().toISOString(),
-			} as UrlWithInformation
+			} as UrlInformation
```
You can remove the "as UrlInformation".

In Typescript, `as X` is used to tell Typescript "I don't care what's in that variable, treat it as type X", which is very blunt. Since you added a type to the variable at the start and it's valid, you won't need that `as`.

Yardenb replied:
Fixed it, thanks!

Yardenb commented:
I rebased my branch to latest version of release/0.2.0, tested it on both relational and in memory storage and I've fixed the last comment I was given.
